sdk/opendp/smartnoise/sql/private_rewriter.py

```diff
@@ -247,45 +247,54 @@ def per_key_clamped(self, query):
         child_scope = Scope()
         relations = query.source.relations
         select = [
-                self.clamp_expression(ne, relations, child_scope, self.options.clamp_columns)
+                self.clamp_expression(ne, relations, child_scope, query, self.options.clamp_columns)
                 for ne in query.select.namedExpressions
             ]
         select = Select(None, select)
         subquery = Query(child_scope.select(), query.source, query.where, None, None, None, None)
         return subquery
 
-    def clamp_expression(self, ne, relations, scope, do_clamp=True):
+    def clamp_expression(self, ne, relations, scope, query, do_clamp=True):
         """
             Lookup the expression referenced in each named expression and
             write a clamped select for it, using the schema
         """
+        def bounds_clamp(colname):
+            if not do_clamp:
+                return None, None
+            if query.agg:
+                grouping_colnames = [col.name for col in query.agg.groupedColumns()]
+                if colname in grouping_colnames:
+                    return None, None
+            minval = None
+            maxval = None
+            sym = col.symbol(relations)
+            if sym.valtype in ["float", "int"] and not sym.unbounded:
+                minval = sym.minval
+                maxval = sym.maxval
+            if minval is None or sym.is_key:
+                return None, None
+            return minval, maxval
+
         exp = ne.expression
         cols = exp.find_nodes(Column)
         if type(exp) is Column:
             cols += [exp]
         for col in cols:
             colname = col.name
-            minval = None
-            maxval = None
-            sym = col.symbol(relations)
-            if do_clamp and sym.valtype in ["float", "int"] and not sym.unbounded:
-                minval = sym.minval
-                maxval = sym.maxval
-                if minval is None or sym.is_key:
```

Comment:
I do not understand this line: why `maxval is None` does not appear here ?

Reply:
That's an oversight, it should be checking maxval.  I just pushed the change